### PR TITLE
Add API to change pages programmatically

### PR DESCRIPTION
**Pull request by irgendeinich**
Adds Android support for the `setPageIndex` API introduced for iOS here: https://github.com/PSPDFKit/react-native/pull/55

### CODE REVIEW

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *
  *   PSPDFKit
  *
- *   Copyright © 2017 PSPDFKit GmbH. All rights reserved.
```
Comment (davidschreiber) [inline android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: Do we have more files with outdated copyright header? We try to keep the original year in here. So it should be updated to `2017-2018`.

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java [hunk 2]
```diff
@@ -13,26 +13,42 @@
...
 
+    @Nullable
+    private Activity currentActivity;
+    @Nullable
+    private Runnable onPdfActivityOpenedTask;
+
     public PSPDFKitModule(ReactApplicationContext reactContext) {
         super(reactContext);
+        if (reactContext.getApplicationContext() instanceof Application) {
+            Application application = (Application) reactContext.getApplicationContext();
```
Comment (davidschreiber) [inline android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: This can lead to class cast exceptions, since some OEMs tend to wrap the application context (which means you won't be able to cast it to your application class).

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java [hunk 3]
```diff
@@ -53,11 +69,84 @@ public void present(@NonNull String document, @NonNull ReadableMap configuration
         }
     }
 
+    @ReactMethod
+    public void setPageIndex(final int pageIndex, final boolean animated) {
```
Comment (davidschreiber) [inline android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: Lines 74-75 introduce a possibility for race (since the `currentActivity` might be replaced between type check and cast). (unless this method runs on the main thread – but it looks like it doesn't)

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java [hunk 4]
```diff
@@ -53,11 +69,84 @@ public void present(@NonNull String document, @NonNull ReadableMap configuration
...
+        if (currentActivity instanceof PdfActivity) {
+            final PdfActivity activity = (PdfActivity) currentActivity;
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    if (activity.getDocument() != null) {
+                        // If the document is loaded we can instantly set the page index.
+                        activity.setPageIndex(pageIndex, animated);
+                    } else {
+                        activity.getPdfFragment().addDocumentListener(new SimpleDocumentListener() {
```
Comment (davidschreiber) [inline android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: Why is this necessary? `PdfFragment` should buffer the page index for as long as no document is loaded.

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java [hunk 5]
```diff
@@ -53,11 +69,84 @@ public void present(@NonNull String document, @NonNull ReadableMap configuration
...
+    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        currentActivity = activity;
```
Comment (davidschreiber) [inline android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: Better call this field `resumedActivity`

File: android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java (hunk 4 above)
Comment (irgendeinich) [reply android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java]: That's true, but unless you disable restoring the last opened page, the last opened page will override any page set before the document was loaded.
